src/python/pants/option/global_options.py

```diff
@@ -487,6 +487,7 @@ class ExecutionOptions:
     local_cache: bool
     process_execution_local_parallelism: int
     process_execution_local_enable_nailgun: bool
+    nailgun_remote_cache_speculation_delay: int
```

Comment:
Because this can be configured on a `Process`-by-`Process` basis, I think that configuring it on a JVM subsystem would be preferable: I'll push a change to do that before merging this.